tests/integ/test_packaging.py

```diff
@@ -966,3 +966,33 @@ def test_get_available_versions_for_packages(session):
     assert returned.keys() == set(packages)
     for key in returned.keys():
         assert len(returned[key]) > 0
+
+
+@pytest.mark.skipif(
+    IS_IN_STORED_PROC,
+    reason="Subprocess calls are not allowed within stored procedures.",
+)
+def test_replicate_local_environment(session):
+    session.custom_package_usage_config = {
+        "enabled": True,
+        "force_push": True,
+    }
+
+    assert not any([package.startswith("cloudpickle") for package in session._packages])
+
+    def naive_add_packages(self, packages):
+        self._packages = packages
+
+    with patch.object(session, "_is_anaconda_terms_acknowledged", lambda: True):
+        with patch.object(Session, "add_packages", new=naive_add_packages):
+            session.replicate_local_environment(
+                ignore_packages={
+                    "snowflake-snowpark-python",
+                    "snowflake-connector-python",
+                    "urllib3",
+                    "tzdata",
+                    "numpy",
+                },
+            )
+
+    assert any([package.startswith("cloudpickle") for package in session._packages])
```

Comment:
can we have more tests for example
- `DEFAULT_PACKAGES` not in `session._packages`
- ignore packages not in `session._packages`